### DISCUSSION TITLE
Open Canvas in Xcode causes Android to cycle restarting App?

### DISCUSSION THREAD

**Discussion by davidakoontz** [skiptools/skip#265] [Q&A] [answered]
Don't know exactly what is causing this...  I think it is me opening a Canvas tool in Xcode... then the running Android App restarts and keeps restarting until I close the Xcode Canvas tool window.  Yeah WERID...

Does NOT happen if I don't use the XCode Canvas window...  But that's no fun!!!

Still an ISSUE for me (10/25/2024)! @marcprux  What should I do - kill this project and start anew?

**Comment by davidakoontz** [comment]
hmm... that was with Skip 1.1.8... let's try updating.... Skip 1.1.11

Get out of the building and restart the world...  

try again.

**Reply by davidakoontz** [reply]
I've had a runaway bunch of gradle commands - Killed them ALL.  Is it safe to go back in???

**Reply by davidakoontz** [reply]
opened the Xcode Canvas and have 16 + gradle command running...
   back to Kill -9 process

**Comment by davidakoontz** [comment]
Here is my Build Phases : Run skip gradle  Shell command.  - I'd really love to get Canvas(previews) working again in Xcode

```
if [ "${SKIP_ZERO}" != "" ]; then
	echo "note: skipping skip due to SKIP_ZERO"
	exit 0
elif [ "${ENABLE_PREVIEWS}" == "YES" ]; then
	echo "note: skipping skip due to ENABLE_PREVIEWS"
	exit 0
elif [ "${ACTION}" == "install" ]; then
	echo "note: skipping skip due to archive install"
	exit 0
else
	SKIP_ACTION="${SKIP_ACTION:-launch}"
fi
PATH=${BUILD_ROOT}/Debug:${BUILD_ROOT}/../../SourcePackages/artifacts/skip/skip/skip.artifactbundle/macos:${PATH}:${HOMEBREW_PREFIX:-/opt/homebrew}/bin
echo "note: running gradle build with: $(which skip) gradle -p ${PWD}/../Android ${SKIP_ACTION:-launch}${CONFIGURATION:-Debug}"
skip gradle -p ../Android ${SKIP_ACTION:-launch}${CONFIGURATION:-Debug}

```

**Comment by marcprux** [comment, marked as answer]
Follow-up in https://github.com/orgs/skiptools/discussions/263.

**Comment by davidakoontz** [comment]
a workaround that might work... Editor > Canvas > Use legacy...

![Screenshot 2024-10-30 at 12 25 42 PM](https://github.com/user-attachments/assets/3060b2f2-9e28-4032-ad45-aa954c22df3e)
